api/src/services/application-csv-export.service.ts

```diff
@@ -131,131 +141,120 @@ export class ApplicationCsvExporterService
               .join(',') + '\n',
           );
 
-          for (let i = 0; i < applications.length / 1000 + 1; i++) {
-            // grab applications 1k at a time
-            const paginatedApplications =
-              await this.prisma.applications.findMany({
-                include: {
-                  ...view.csv,
-                  demographics: queryParams.includeDemographics
-                    ? {
-                        select: {
-                          id: true,
-                          createdAt: true,
-                          updatedAt: true,
-                          ethnicity: true,
-                          gender: true,
-                          sexualOrientation: true,
-                          howDidYouHear: true,
-                          race: true,
-                        },
+          const promiseArray: Promise<string>[] = [];
+          for (let i = 0; i < applications.length; i += NUMBER_TO_PAGINATE_BY) {
+            promiseArray.push(
+              new Promise(async (resolve) => {
+                // grab applications NUMBER_TO_PAGINATE_BY at a time
+                const paginatedApplications =
+                  await this.prisma.applications.findMany({
+                    include: {
+                      ...view.csv,
+                      demographics: queryParams.includeDemographics
+                        ? {
+                            select: {
+                              id: true,
+                              createdAt: true,
+                              updatedAt: true,
+                              ethnicity: true,
+                              gender: true,
+                              sexualOrientation: true,
+                              howDidYouHear: true,
+                              race: true,
+                            },
+                          }
+                        : false,
+                    },
+                    where: {
+                      listingId: queryParams.listingId,
+                      deletedAt: null,
+                    },
+                    skip: i,
```

Comment:
Curious why this changed from (i * num_to_paginate_by) to just i

Reply:
ah yeah that has to do with how I was iterating earlier 
in the for loop I was trying to count by batch number basically (0,1,2,..n) but I realized its easier to just count by where we are in the batch (0, 500, 1k, 1.5k, n) instead so we don't need to figure out how many to skip ourselves anymore